base/arrayshow.jl

```diff
@@ -101,7 +101,21 @@ function print_matrix_row(io::IO,
         if isassigned(X,Int(i),Int(j)) # isassigned accepts only `Int` indices
             x = X[i,j]
             a = alignment(io, x)
-            sx = sprint(show, x, context=io, sizehint=0)
+
+            # If compact is set, keep the value, else set to true
+            compact = get(io, :compact, nothing)
+            if compact === nothing
+                compact = true
+            end
+            context = IOContext(io, :compact=>compact)
```

Comment:
I think this can be removed and we can just keep the `:compact` logic in `show(io::IO, ::MIME"text/plain", X::AbstractArray)`.

Reply:
Just to double check, are you suggesting I move that logic to `show(io::IO, ::MIME"text/plain", X::AbstractArray)`? or just deleting this logic and keeping the logic that is currently in that function:

```julia
    if !haskey(io, :compact) && length(axes(X, 2)) > 1
        io = IOContext(io, :compact => true)
    end
```

Reply:
Keep what's currently in that function.